pilot/pkg/config/kube/crd/controller/client.go

```diff
@@ -370,13 +377,27 @@ func (cl *Client) List(kind resource.GroupVersionKind, namespace string) ([]mode
 		obj, err := crd.ConvertObject(s, item, cl.domainSuffix)
 		if err != nil {
 			errs = multierror.Append(errs, err)
-		} else {
+		} else if cl.objectInEnvironment(obj) {
 			out = append(out, *obj)
 		}
 	}
 	return out, errs
 }
 
+func (cl *Client) objectInEnvironment(o *model.Config) bool {
```

Comment:
You might also be able to filter in the k8s list/watch. This would reduce memory usage in the client cache and decrease load on the kube-apiserver.

Reply:
I really wanted to, but there is no `OR` support on labels. So we cannot express `label is empty OR label is revision123`. Unless there is some tricky boolean logic I missed

Reply:
"label is empty" or "label not set at all"? istio.io/revision is only added if the value is not empty. 

Assuming "label is not set at all", then the "OR" decision is made when the informer is started, i.e. watch all objects vs. only objects with istio.io/rev=<revision>.

Reply:
"label not set at all". If its not set, it applies to all revisions. This is the "normal" case  - all current configs are likely in this category. If we just watch objects with istio.io/rev label then we miss the "global" objects